### PR TITLE
Add health analyzer and medical scanner ECS

### LINKED ISSUE #5602: Sentinent monkeys can be cloned into humans.
## Description

How to replicate:

- Spawn a bunch of monkeys. I think they become ghost roles automatically on spawn, but I'm not sure on that.
- Wait for the monkeys to be inhabited by bored ghosts. 
- Take a sentient monkey to a medical scanner. They'll go into the scanner the same way a human does, and the entry in the cloner will read "monkey".
- Kill the monkey, then clone them. They will clone into whoever they were before they went ghost role. 
- Wonder if we're all just really sentient monkeys at heart.

### PR DESCRIPTION
**Pull request by Fishfish458**
## About the PR 

Purpose of this is both to implement Health Analyzer and as a stepping stone to implementing some changes to the cloner.

Once cloning console system and component are integrated the medical scanner should have no UI and basically just be a mostly dumb container that holds the player entity.

This should also fix #5602 by not letting any mob without a humanoid appearance component be scanned

**Screenshots**

health analyzer

scanning

crafting

added

**Changelog**

:cl:
- add: Health Analyzers have now been added to various medical containers and the autolathe

### CODE REVIEW

File: Content.Shared/MedicalScanner/SharedHealthAnalyzerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                bool? isAlive,
+                string? totalDamage,
+                List<MobDamageGroup> damageGroups)
+            {
+                TargetName = targetName;
+                IsAlive = isAlive;
+                TotalDamage = totalDamage;
+                DamageGroups = damageGroups;
+            }
+        }
```
Comment (metalgearsloth) [inline Content.Shared/MedicalScanner/SharedHealthAnalyzerComponent.cs:26]: * Target name not needed.
* * IsAlive probably not needed and can be moved to shared.  
* Realistically you could probably just have this contain an EntityUid and then have the client handle it all. This also means you don't need the mobdamagegroup and it cuts down on the state's bandwidth a lot.

File: Content.Shared/MedicalScanner/SharedHealthAnalyzerComponent.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            public MobDamageGroup(string? groupName, string? groupTotalDamage, Dictionary<string, string>? groupedMinorDamages)
+            {
+                GroupName = groupName;
+                GroupTotalDamage = groupTotalDamage;
+                GroupedMinorDamages = groupedMinorDamages;
+            }
+        }
+
+        [Serializable, NetSerializable]
+        public enum HealthAnalyzerUiKey
```
Comment (metalgearsloth) [inline Content.Shared/MedicalScanner/SharedHealthAnalyzerComponent.cs:43]:
```suggestion
        public enum HealthAnalyzerUiKey : byte
```

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Okay I did my review before I checked that client does get damageablecomponent data. For now it'll be fine until interaction prediction PR in + do_afters predicted but the state should just be handled by the client instead IMO.

File: Content.Shared/MedicalScanner/SharedHealthAnalyzerComponent.cs (hunk 1 above)
Comment (Fishfish458) [reply Content.Shared/MedicalScanner/SharedHealthAnalyzerComponent.cs:26]: Going to opt for this,  only passing in an entityuid and getting the damageable and name on the client.
The only downside is that there the info isn't saved between closing and opening the UI but it's definitely a lot cleaner.

File: Content.Server/Medical/MedicalScannerSystem.cs [hunk 3]
```diff
@@ -1,42 +1,90 @@
...
-namespace Content.Server.Medical
+using static Content.Shared.MedicalScanner.SharedMedicalScannerComponent;
+
+namespace Content.Server.MedicalScanner
 {
     [UsedImplicitly]
     internal sealed class MedicalScannerSystem : EntitySystem
     {
         [Dependency] private readonly ActionBlockerSystem _actionBlockerSystem = default!;
         [Dependency] private readonly IGameTiming _gameTiming = default!;
```
[the remark this answers is not in the document]
Comment (Fishfish458) [reply Content.Server/Medical/MedicalScannerSystem.cs]: I've deleted the gametiming which is what I hope it meant, it wasn't being used.

**Comment by Fishfish458** [comment]
So, I've gone through and changed:
Health analyzer interface now only receives the entity uid of whoever was scanned and fills out the rest itself
added cancellation token to health analyzer.
removed on component init from health analyzer
added break on stun and break on hand change to health analyzer
removed the occupant name from the medical scanner, that'll be irrelevant with the cloning console anyhow.
changed the medical scanner to now always use the occupant sprite when it is occupied.
added a small update delay to medical scanner since it shouldn't need to be every frame.
various fixes that I stupidly missed.

**Pull request merged**